python/tvm/relay/op/_tensor_grad.py

```diff
@@ -472,8 +472,8 @@ def bias_add_grad(orig, grad):
 def dense_grad(orig, grad):
     """Returns [grad' @ weight, data @ grad']"""
     data, weight = orig.args
-    return [collapse_sum_like(transpose(grad) * weight, data),
-            collapse_sum_like(data * transpose(grad), weight)]
+    return [collapse_sum_like(_nn.dense(grad, transpose(weight)), data),
```

Comment:
Please add units arg too to support batching.

``
return [collapse_sum_like(_nn.dense(grad, transpose(weight), units=weight.checked_type.shape[1]), data),
    collapse_sum_like(_nn.dense(transpose(grad), transpose(data), units=data.checked_type.shape[1]), weight)]
``

Reply:
Thank you for your response!
After correcting it, I found that both x86 and CUDA only support dense without batching, I tried testing with arm-cpu(mobile) but came into [python/tvm/relay/op/strategy/x86.py](https://github.com/apache/incubator-tvm/blob/master/python/tvm/relay/op/strategy/x86.py) and failed as well, so could you please tell me how to run dense with batching? 
Thanks again!

Reply:
In above example of data (5, 4) and weight (3, 4) implies a dense with 4 inputs and yielding 3 outputs each for 5 units. Hence 5 here is the batch when we apply to a network.

We very well support units/batches. Can you share details on the error you got while units is added as arg ?